### DISCUSSION TITLE
brew for 2022 unatteneded

### DISCUSSION THREAD

**Discussion by Francou2021** [Homebrew/discussions#3447] [Getting started]
I am trying to setup unattened brew install 
every link is old and gets the error Warning: The Ruby Homebrew installer is now deprecated and has been rewritten in
Bash. Please migrate to the following command:
  /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"

But I can not figure anyway to run that as unattended  when I do an echo password | sudo  /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
give error dont run this as root

**Comment by SMillerDev** [comment]
You need to use `SUDO_ASKPASS`

**Reply by Francou2021** [reply]
sorry but II might be wrong on this but does not mean you have add user to sudouser which means user interferance 
I need to ensure to install brew without interfernce all coomands I found with google has been deprecated or need sudo.

**Reply by SMillerDev** [reply]
You will always need sudo because the brew install script needs sudo.

**Reply by Francou2021** [reply]
Let me clarify I need way to bypass sodu password  how do I set the command which allows this so there no user interfereenace

**Reply by SMillerDev** [reply]
Look up how to use `SUDO_ASKPASS`

**Reply by Francou2021** [reply]
stratght from apple. this means user inrefernce and I try to make unatteneded 

From the sudoers file.
Use visudo to add the following line to the sudoers file:

Defaults:ALL askpass=/usr/local/bin/pw

Using the SUDO_ASKPASS environment variable.
export SUDO_ASKPASS=/usr/local/bin/pw

This might work too (assuming SUDO_ASKPASS has been previously exported):

SUDO_ASKPASS=/usr/local/bin/pw sudo -A command

**Comment by carlocab** [comment]
Use `SUDO_ASKPASS` and then run the install script.

Note: you should _not_ set `SUDO_ASKPASS` to your password. That's not how it works.

**Reply by Francou2021** [reply]
how do I instal;l this on mac since it's not native

**Reply by carlocab** [reply]
`SUDO_ASKPASS` is an environment variable. You do not need to install it. See `man sudo`.

**Reply by Francou2021** [reply]
as stated, prior then this has do with the sudo file make user interference 

stratght from apple

1. From the sudoers file.



Use visudo to add the following line to the sudoers file:



Defaults:ALL    askpass=/usr/local/bin/pw



2. Using the SUDO_ASKPASS environment variable.



export SUDO_ASKPASS=/usr/local/bin/pw



This might work too (assuming SUDO_ASKPASS has been previously exported):



SUDO_ASKPASS=/usr/local/bin/pw sudo -A command

**Reply by carlocab** [reply]
You don't need to edit your `sudoers` file. You just need to use `SUDO_ASKPASS` correctly.

If you want, you can create a file called `mypassword` that contains
```
#!/bin/sh
echo "insert-your-password-here"
```
and make `mypassword` executable. Then, do
```
export SUDO_ASKPASS=/path/to/mypassword
```
You're also free to edit your `sudoers` file if you wish, but the right configuration is tricky and you could break `sudo` by doing so.

**Reply by Francou2021** [reply]
So are you stated create a file then put into a folder then run sudo_pass because that sounds like user interfernce 
I want dto do this  autmatically and remotely

**Reply by SMillerDev** [reply]
If you don't know how to automate creating files I think that's a big step beyond the level of support I'm willing to provide here.

**Reply by Francou2021** [reply]
Why I need to create this file on all macs if I have 1000s macs that means its unscable 

Why cannot I not have an command that I run with sudo password pass, and it simply works
